packages/next/client/link.tsx

```diff
@@ -40,6 +40,7 @@ function prefetch(
   as: string,
   options?: PrefetchOptions
 ): void {
+  if (!router) return
   if (typeof window === 'undefined') return
```

Comment:
You can do something like this instead to save some bytes

```suggestion
  if (typeof window === 'undefined' || !router) return
```